apps/server/src/modules/tool/uc/external-tool-configuration.uc.ts

```diff
@@ -61,61 +66,65 @@ export class ExternalToolConfigurationUc {
 				schoolId,
 			}),
 			this.contextExternalToolService.findContextExternalTools({
-				context: {
-					id: contextId,
-					type: contextType,
-				},
+				context: { id: contextId, type: contextType },
 			}),
 		]);
 
-		const schoolExternalToolsInUse: SchoolExternalToolDO[] = this.filterForSchoolExternalToolsInUse(
+		const availableSchoolExternalTools: SchoolExternalToolDO[] = this.filterForAvailableSchoolExternalTools(
 			schoolExternalTools,
 			contextExternalToolsInUse
 		);
 
-		const toolIdsInUse: EntityId[] = schoolExternalToolsInUse.map(
-			(schoolExternalTool: SchoolExternalToolDO): EntityId => schoolExternalTool.toolId
-		);
-
-		const availableTools: ExternalToolDO[] = this.filterForAvailableExternalTools(
+		const availableToolsForContext: AvailableToolsForContext[] = this.filterForAvailableExternalTools(
 			externalTools.data,
-			schoolExternalTools,
-			toolIdsInUse
+			availableSchoolExternalTools
 		);
 
-		return availableTools;
+		return availableToolsForContext;
 	}
 
-	private filterForSchoolExternalToolsInUse(
+	private filterForAvailableSchoolExternalTools(
 		schoolExternalTools: SchoolExternalToolDO[],
 		contextExternalToolsInUse: ContextExternalToolDO[]
 	): SchoolExternalToolDO[] {
-		const schoolExternalToolsInUse: SchoolExternalToolDO[] = schoolExternalTools.filter(
+		const availableSchoolExternalTools: SchoolExternalToolDO[] = schoolExternalTools.filter(
 			(schoolExternalTool: SchoolExternalToolDO): boolean => {
 				const hasContextExternalTool: boolean = contextExternalToolsInUse.some(
 					(contextExternalTool: ContextExternalToolDO) =>
 						contextExternalTool.schoolToolRef.schoolToolId === schoolExternalTool.id
 				);
 
-				return hasContextExternalTool;
+				return !hasContextExternalTool;
 			}
 		);
 
-		return schoolExternalToolsInUse;
+		return availableSchoolExternalTools;
 	}
 
 	private filterForAvailableExternalTools(
 		externalTools: ExternalToolDO[],
-		schoolExternalTools: SchoolExternalToolDO[],
-		toolIdsInUse: EntityId[]
-	): ExternalToolDO[] {
-		const availableTools: ExternalToolDO[] = externalTools.filter((tool: ExternalToolDO): boolean => {
-			const hasSchoolExternalTool: boolean = schoolExternalTools.some(
-				(schoolExternalTool: SchoolExternalToolDO): boolean => schoolExternalTool.toolId === tool.id
-			);
-
-			return !tool.isHidden && !!tool.id && !toolIdsInUse.includes(tool.id) && hasSchoolExternalTool;
-		});
+		availableSchoolExternalTools: SchoolExternalToolDO[]
+	): AvailableToolsForContext[] {
+		const toolsWithSchoolTool: (AvailableToolsForContext | undefined)[] = availableSchoolExternalTools.map(
+			(schoolExternalTool: SchoolExternalToolDO) => {
+				const externalTool: ExternalToolDO | undefined = externalTools.find(
+					(tool: ExternalToolDO) => schoolExternalTool.toolId === tool.id
+				);
+
+				if (!externalTool) {
+					return undefined;
```

Comment:
i would prefer a return null. If you return null it is visible for other that is a desired behavior